webrtc/RTCPeerConnection-helper.js

```diff
@@ -609,13 +614,40 @@ const trackFactories = {
   }
 };
 
+// Get the signal from a video element inserted by createNoiseStream
+function getVideoSignal(v) {
+  if (v.videoWidth < 21 || v.videoHeight < 21) {
+    return null;
+  }
+  const canvas = new OffscreenCanvas(v.videoWidth, v.videoHeight);
```

Comment:
@alvestrand is it necessary to use `OffscreenCanvas` here? In https://wpt.fyi/results/webrtc/RTCRtpSender-replaceTrack.https.html?label=pr_head&max-count=1&pr=22779 it looks like Firefox and Safari fail this test because of this. Could a regular canvas be used just as well?

Reply:
Even if `canvas` is used the expectation of the test is not clear where the codec used is lossy: why is a specific pixel expected to be in an exact coordinate when using a lossy video codec.

The timeout is probably related to `loadedmetadata` never being fired in the code in this PR see https://github.com/w3c/webrtc-pc/issues/2506#issuecomment-618206122

Reply:
`detectSignal` allows for an error margin of 1, but good point that this may not always hold.

I don't see a timeout in the [test results](https://wpt.fyi/results/webrtc/RTCRtpSender-replaceTrack.https.html?label=pr_head&max-count=1&pr=22779) so there isn't anything to investigate there I think.

Reply:
The test paints a box in a specific color, and then checks that the middle of that box is the expected color (or close to it). The margin should be big enough that even edge-blurring codecs don't mess things up; if the video width and height are the same (no resize), the box should be in the same place.

The amount of color disparity one can expect should be tested; good point. Later CL.